### PR TITLE
Added S3 variables and parameters

### PR DESCRIPTION
**Pull request by deanp3it**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -24,8 +24,23 @@ variable "event_types" {
 
 variable "lambda" {
   description = "Map of any additional arguments for the upstream lambda module. See <https://github.com/terraform-aws-modules/terraform-aws-lambda>"
-  type        = any
-  default     = {}
+  type = object({
```
Comment (lorengordon) [inline variables.tf:27]: There are several attributes here that aren't actually being used in the code. Not sure I've caught all of them, but I'm seeing at least, `build_in_docker`, `memory_size`, `runtime`, and `timeout`.

**Review by lorengordon — CHANGES_REQUESTED** [review]
One comment. Other than that, should be good to go.

File: variables.tf (hunk 1 above)
Comment (deanp3it) [reply variables.tf:27]: I believe all of the unused attributes have been removed.

**Review by lorengordon — APPROVED** [review]
🚀

**Pull request merged**